### ISSUE TITLE
`getPrice()` returns a string when setting custom price in admin order

### ISSUE THREAD

**Issue by ghost** [magento/magento2#25968]
### Preconditions
M 2.3.3
### Steps to reproduce
1. The method `getPrice()` from `/vendor/magento/module-sales/Model/Order/Item.php`

```
    /**
     * Return price
     *
     * @return float|null
     */
    public function getPrice()
    {
        return $this->getData(OrderItemInterface::PRICE);
    }
```

### Expected result
Clearly states it should return a float, or null.
### Actual result
However, when creating a MOTO order, and you set a custom price for a product, when you call `getPrice()` for the order items, the returned value will be a string, not a float.

We switched to `getBasePrice()` (seemingly operates the same as `getPrice()`....) as that should also return a float, or null. It worked fine locally, but had a report that on Magento Cloud, it would return a string.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Echo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-29645` were created

**Issue Available**: @engcom-Echo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Unfortunately, we are archiving this ticket now as it did not get much attention from both Magento Community and Core developers for an extended period. This is done in an effort to create a quality, community-driven backlog which will allow us to allocate the required attention more easily.

Please feel free to comment or reopen according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines) 
the ticket if you are still facing this issue on the latest `2.x-develop` branch. Thank you for collaboration.

**Comment by ghost** [comment]
@magento-engcom-team Are you serious?

> Unfortunately, we are archiving this ticket now as it did not get much attention from both Magento Community and Core developers for an extended period

I opened this ticket yesterday....

**Comment by AndreyChorniy** [comment]
@paul-gene This is mistake. I am going to working on this ticket.

**Comment by ghost** [comment]
@AndreyChorniy please just give up adding labels. If "good first issue" is in relation to me, then it's certainly not true, I've reported other issues in the past.

You're fighting a losing battle with the bots here.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @sdzhepa 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-29670` were created

**Issue Available**: @sdzhepa, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by sdzhepa** [comment]
Hello @paul-gene 

Thank you for contribution and collaboration!

Sorry for  https://github.com/magento/magento2/issues/25968#issuecomment-564555318 and https://github.com/magento/magento2/issues/25968#issuecomment-564556030 bot closed this issue accidentally by mistake due to delays and queues through all integration logic between GitHub and internal Jira...

I have returned the Confirmation label.

**Comment by VladimirZaets** [comment]
Hi @paul-gene. Thank you for your report. 
The issue has been fixed in magento/magento2#26313 by @AndreyChorniy in 2.4-develop branch
Related commit(s):
 - [0b7a99d7a330216a829056af1b7eb876c14657db](https://github.com/magento/magento2/commit/0b7a99d7a330216a829056af1b7eb876c14657db)
 - [28e8e5aa46ffe2cfed80124bb95d3ce025a127c8](https://github.com/magento/magento2/commit/28e8e5aa46ffe2cfed80124bb95d3ce025a127c8)
 - [71da244811484747f55bf3ccb1244844ff903fd6](https://github.com/magento/magento2/commit/71da244811484747f55bf3ccb1244844ff903fd6)

The fix will be available with the upcoming 2.4.0 release.
